### PR TITLE
Ast parser and tree regex

### PR DESCRIPTION
**Pull request by zsr2531**

### CODE REVIEW

**Comment by Washi1337** [comment]
Linked to #50

**Comment by zsr2531** [comment]
Linked to #51

File: src/Core/Echo.Ast/InstructionExpression.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+using Echo.Core.Graphing;
+
+namespace Echo.Ast
+{
+    /// <summary>
+    /// Represents an instruction in the AST
+    /// </summary>
+    public sealed class InstructionExpression<TInstruction> : Expression<TInstruction>
+    {
+        /// <summary>
```
Comment (Washi1337) [inline src/Core/Echo.Ast/InstructionExpression.cs:13]: Add constructor for expressions without arguments, and a `params Expression<TInstruction>[]` for consistency and readability.

**Review by Washi1337 — CHANGES_REQUESTED** [review]

File: src/Core/Echo.ControlFlow/Serialization/Dot/ControlFlowNodeAdorner.cs [hunk 2]
```diff
@@ -15,7 +15,7 @@ public class ControlFlowNodeAdorner<TInstruction> : IDotNodeAdorner
         /// Creates a new <see cref="ControlFlowNodeAdorner{TInstruction}"/> with the default formatter.
         /// </summary>
         public ControlFlowNodeAdorner()
-            : this(DefaultInstructionFormatter<TInstruction>.Instance) { }
```
[the remark this answers is not in the document]
Comment (zsr2531) [reply src/Core/Echo.ControlFlow/Serialization/Dot/ControlFlowNodeAdorner.cs]: No idea, I forgot to be honest :D

**Review by Washi1337 — APPROVED** [review]

**Pull request merged**